File: BJ/BjLevel/20_Binary_search/1920.cpp
#include <bits/stdc++.h>

using namespace std;

void binarysearch(vector<int>& arr, int target, int start, int end) {
	while (start <= end) {
		int mid = (start+end)/2;
		if (arr[mid] == target) {
			printf("1\n");
			return;
		}
		else if (arr[mid] < target) start = mid + 1;
		else end = mid - 1;
	}
	printf("0\n");
}

int n, m;
vector<int> arr;

int main() {
	cin >> n;
	for (int i=0; i!=n; i++) {
		int num;
		scanf("%d", &num);
		arr.push_back(num);
	}
	sort(arr.begin(), arr.end());
	cin >> m;
	for (int i=0; i!=m; i++) {
		int find_num;
		scanf("%d", &find_num);
		binarysearch(arr,find_num,0,n);
	}
	return 0;
}

File: BJ/01000/1920.py
n = int(input())
arrN = list(map(int, input().split()))
m = int(input())
arrM = list(map(int, input().split()))

def binary_search(arr, num, start, end):
    if start > end:
        return None
    else:
        mid = (start + end)//2
        if arr[mid] == num:
            return mid
        elif arr[mid] > num:
            return binary_search(arr, num, start, mid-1)
        else:
            return binary_search(arr, num, mid+1, end)
            
# def binary_search(array, target, start, end):
#     while (start <= end):
#         mid = (start + end) // 2
#         if array[mid] == target:
#             return mid
#         elif array[mid] > target:
#             end = mid - 1
#         else:
#             start = mid + 1
#     return None



arrN.sort()
for num in arrM:
    if binary_search(arrN, num, 0, len(arrN)-1) != None:
        print(1)
    else:
        print(0)